pcdsdevices/pulsepicker.py

```diff
@@ -0,0 +1,155 @@
+    states_list = ['OPEN', 'CLOSED']
+    in_states = ['CLOSED']
+    out_states = ['OPEN']
+    _states_alias = {'CLOSED': ['CLOSED', 'IN']}
+    _state_logic = {'blade': {0: 'OPEN',
+                              1: 'CLOSED',
+                              2: 'CLOSED'}}
+
+    _default_config_attrs = ['mode']
+
+    def _do_move(self, state):
+        """
+        Handle move requests for basic open/close commands. This allows us to
+        make calls like pulsepicker.move('OPEN')
+        """
+        if state.name == 'OPEN':
+            self.open(wait=False)
+        if state.name == 'CLOSED':
+            self.close(wait=False)
+
+    def _wait(self, sig, *goals):
+        """
+        Helper function to wait for a signal to reach a value. This is used
+        here because most commands are only valid when mode is IDLE.
+        """
+        def cb(value, *args, **kwargs):
+            logger.debug((value, goals))
+            return value in goals
+        status = SubscriptionStatus(sig, cb)
+        status_wait(status)
+
+    def _log_request(self, mode):
+        logger.debug('Request %s %s', self.name, mode)
+
+    def reset(self, wait=False):
+        """
+        Cancel the current mode.
+        """
+        self._log_request('RESET')
+        self.cmd_reset.put(1)
+        if wait:
+            self._wait(self.mode, 0, 'IDLE')
+
+    def open(self, wait=False):
+        """
+        Cancel the current mode and leave the PulsePicker OPEN.
+        """
+        self.reset(wait=True)
+        self._log_request('OPEN')
+        self.cmd_open.put(1)
+        if wait:
+            self._wait(self.blade, 0, 'OPEN')
+
+    def close(self, wait=False):
+        """
+        Cancel the current mode and leave the PulsePicker CLOSED.
+        """
+        self.reset(wait=True)
+        self._log_request('CLOSED')
+        self.cmd_close.put(1)
+        if wait:
+            self._wait(self.blade, 1, 2, 'CLOSED -', 'CLOSED +')
+
+    def flipflop(self, wait=False):
+        """
+        Change the current mode to FLIP-FLOP.
+        """
+        self.reset(wait=True)
+        self._log_request('FLIP-FLOP')
+        self.cmd_flipflop.put(1)
+        if wait:
+            self._wait(self.mode, 2, 'FLIP-FLOP')
+
+    def burst(self, wait=False):
+        """
+        Change the current mode to BURST.
+        """
+        self.reset(wait=True)
+        self._log_request('BURST')
+        self.cmd_burst.put(1)
+        if wait:
+            self._wait(self.mode, 3, 'BURST')
+
+    def follower(self, wait=False):
+        """
+        Change the current mode to FOLLOWER.
+        """
+        self.reset(wait=True)
+        self._log_request('FOLLOWER')
+        self.cmd_follower.put(1)
+        if wait:
+            self._wait(self.mode, 6, 'FOLLOWER')
+
+
+class PulsePickerInOut(PulsePicker):
+    """
+    PulsePicker paired with a states record to control the Y position. This
+    allows us to insert and remove the entire device from the beam.
```

Comment:
Maybe add a note that the reason this isn't all one class is because these PVs come from a separate IOC

Reply:
I will add a note, with some instructions on how to set it up maybe.